### PR TITLE
Mail service

### PR DESCRIPTION
**Pull request by Skava600**

### CODE REVIEW

File: RecordHub.MailService/RecordHub.MailService.Api/Buffer-20230623.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
...nt, ArrayPool`1 pool, Func`2 socketFactory, TimeSpan connectionTimeout, TimeSpan readTimeout, TimeSpan writeTimeout)\n   at RabbitMQ.Client.ConnectionFactory.CreateFrameHandler(AmqpTcpEndpoint endpoint)\n   at RabbitMQ.Client.EndpointResolverExtensions.SelectOne[T](IEndpointResolver resolver, Func`2 selector)","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":null,"HResult":-2146233088,"Source":"RabbitMQ.Client","WatsonBuckets":null}]},"HelpURL":null,"StackTraceString":"   at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName)\n   at RabbitMQ.Client.ConnectionFactory.CreateConnection(IList`1 hostnames, String clientProvidedName)\n   at MassTransit.RabbitMqTransport.ConnectionContextFactory.CreateConnection(ISupervisor supervisor) in /_/src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/ConnectionContextFactory.cs:line 86","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":null,"HResult":-2146232800,"Source":"RabbitMQ.Client","WatsonBuckets":null},"messageTemplate":"Connection Failed: {InputAddress}","properties.InputAddress":"rabbitmq://messagebus/","logger":"MassTransit"}
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Api/Buffer-20230623.txt]: You should better configure gitignore)

File: RecordHub.MailService/RecordHub.MailService.Api/Middleware/GlobalExceptionHandlerMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace RecordHub.MailService.Api.Middleware
+{
+    public class GlobalExceptionHandlerMiddleware
+    {
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Api/Middleware/GlobalExceptionHandlerMiddleware.cs]: TODO?)

File: RecordHub.MailService/RecordHub.MailService.Api/Program.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using Hangfire;
+using RecordHub.MailService.Api.Extensions;
+using RecordHub.MailService.Infrastructure;
+using RecordHub.MailService.Infrastructure.Config;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Configuration.SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json")
+                .Build();
+
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Api/Program.cs:10]: many empty lines

File: RecordHub.MailService/RecordHub.MailService.Api/Program.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+builder.Services.ConfigureMail(builder.Configuration);
+builder.Services.AddInfrastructureServices(builder.Configuration);
+builder.Services.AddMassTransit();
+builder.ConfigureSerilog();
+var app = builder.Build();
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Api/Program.cs:16]: Maybe just a new line before Build()?

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Events;
+
+namespace RecordHub.MailService.Infrastructure.Config
+{
+    public static class LoggingConfiguration
+    {
+
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs]: remove line

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        public static void ConfigureSerilog(this WebApplicationBuilder builder)
+        {
+            builder.Logging.ClearProviders();
+            LogEventLevel logLevel;
+            if (builder.Environment.IsProduction())
+                logLevel = LogEventLevel.Information;
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs:17]: brackets

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            else
+                logLevel = LogEventLevel.Debug;
+            var log = new LoggerConfiguration()
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Config/LoggingConfig.cs:24]: line after if/else/while etc.

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Consumers/SendEmailConsumer.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
+﻿using Hangfire;
+using MassTransit;
+using RecordHub.MailService.Application.Services;
+using RecordHub.Shared.Models;
+
+namespace RecordHub.MailService.Infrastructure.Consumers
+{
+    public class SendEmailConsumer : IConsumer<MailData>
+    {
+        private readonly IMailService mailService;
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Consumers/SendEmailConsumer.cs]: Wrong private field name convention

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/ServiceExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        public static IServiceCollection AddMassTransit(this IServiceCollection services)
+        {
+            services.AddMassTransit(x =>
+            {
+                x.AddConsumer<SendEmailConsumer>();
+                x.UsingRabbitMq((context, cfg) =>
+                {
+                    cfg.Host("messagebus", "/", h =>
+                    {
+                        h.Username("guest");
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/ServiceExtensions.cs]: appsettings

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Services/MailService.cs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+            var mail = new MimeMessage();
+
+            #region Sender / Receiver
+            // Sender
+            mail.From.Add(new MailboxAddress(_settings.DisplayName, mailData.From ?? _settings.From));
+            mail.Sender = new MailboxAddress(mailData.DisplayName ?? _settings.DisplayName, mailData.From ?? _settings.From);
+
+            // Receiver
+            foreach (string mailAddress in mailData.To)
+                mail.To.Add(MailboxAddress.Parse(mailAddress));
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Services/MailService.cs:32]: also brackets

File: RecordHub.MailService/RecordHub.MailService.Infrastructure/Services/MailService.cs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                await smtp.ConnectAsync(_settings.Host, _settings.Port, SecureSocketOptions.StartTls, ct);
+            }
+            await smtp.AuthenticateAsync(_settings.UserName, _settings.Password, ct);
+            await smtp.SendAsync(mail, ct);
+            await smtp.DisconnectAsync(true, ct);
+
+            #endregion
+
+            return true;
```
Comment (Y45K0VL3V) [inline RecordHub.MailService/RecordHub.MailService.Infrastructure/Services/MailService.cs]: Does false possible?

**Review by Y45K0VL3V — CHANGES_REQUESTED** [review]

File: RecordHub.MailService/RecordHub.MailService.Api/Middleware/GlobalExceptionHandlerMiddleware.cs (hunk 2 above)
Comment (Skava600) [reply RecordHub.MailService/RecordHub.MailService.Api/Middleware/GlobalExceptionHandlerMiddleware.cs]: Hah yeah, I forgot)

**Review by Y45K0VL3V — APPROVED** [review]

**Pull request merged**